### DISCUSSION TITLE
Ichimoku - cloud: How to fill the space between two line series?

### DISCUSSION THREAD

**Discussion by UweR70** [tradingview/lightweight-charts#1369] [General]
Imagine two line series A and B.
How to fill the space between these two lines?

Please notice that in my case the two line series are part of the calculated Ichimoku data.
Means one array item contains all needed data, like for example for "array item # 47":

`result.indicatorResult[47].time = 1682920800`
`result.indicatorResult[47].senkouSpanA = 0.07447`
`result.indicatorResult[47].senkouSpanB = 0.07386`
`result.indicatorResult[47]. ...`

I believe the important code part should look like this NOT working code:

`const dataCloudGreen = result`
`  .indicatorResult`
`  .filter(item =>`
`	item.senkouSpanA != null`
`	&& item.senkouSpanB != null`
`	&& item.senkouSpanA > item.senkouSpanB)`
`  .map(item => {`
`	return {`
`	  time: item.time,`
`	  value: item.senkouSpanA,`
`	  low: item.senkouSpanB,`
`	};`
`});`

It does not work because there is no "low", nor "value2" property.
Also this does NOT work:

`return {`
`  time: item.time,`
`  value: {`
`	high: item.senkouSpanA,`
`    low: item.senkouSpanB,`
`  }`
`};`

Any ideas?

**Comment by UweR70** [comment]
So now answer ... i´ve developed it ... and found some issues in Binance:

![x](https://raw.githubusercontent.com/UweR70/Images/main/chart/001_Binance.png)

my solution 

![y](https://raw.githubusercontent.com/UweR70/Images/main/chart/002_My_solution.png)

![](https://raw.githubusercontent.com/UweR70/Images/main/chart/003_Complete_chart.png)

![](https://raw.githubusercontent.com/UweR70/Images/main/chart/004_details_1.png)

![](https://raw.githubusercontent.com/UweR70/Images/main/chart/005_details_2.png)


chart start:

![](https://raw.githubusercontent.com/UweR70/Images/main/chart/006_chart_start.png)

chart end: (BTW: my ichimoku implementation contains the correct number of data points in the future - means 26 and not 25 or something else - and of course it can be configured completely dynamically)

![](https://raw.githubusercontent.com/UweR70/Images/main/chart/007_chart_end.png)

**Comment by UweR70** [comment]
The current state:
Implemented both: Cloud handling on client as also server side.

![](https://raw.githubusercontent.com/UweR70/Images/main/chart/010_Sa_May_27_2023__current_state.png)

![](https://raw.githubusercontent.com/UweR70/Images/main/chart/011_Sa_May_27_2023__current_state.png)

![](https://raw.githubusercontent.com/UweR70/Images/main/chart/012_Sa_May_27_2023__current_state.png)

**Reply by difof** [reply]
Can you elaborate on how you filled the area?
